### DISCUSSION TITLE
java.lang.NoClassDefFoundError: javax/xml/bind/annotation/XmlSchema

### DISCUSSION THREAD

**Discussion by mattiascibien** [microsoft/openjdk#42] [Q&A] [answered]
When running `maui-check` on windows with the Microsoft build of OpenJDK I get the following error when trying to create an emulator:

```
Exception in thread "main" java.lang.NoClassDefFoundError: javax/xml/bind/annotation/XmlSchema
        at com.android.repository.api.SchemaModule$SchemaModuleVersion.<init>(SchemaModule.java:156)
        at com.android.repository.api.SchemaModule.<init>(SchemaModule.java:75)
        at com.android.sdklib.repository.AndroidSdkHandler.<clinit>(AndroidSdkHandler.java:81)
        at com.android.sdklib.tool.AvdManagerCli.run(AvdManagerCli.java:213)
        at com.android.sdklib.tool.AvdManagerCli.main(AvdManagerCli.java:200)
Caused by: java.lang.ClassNotFoundException: javax.xml.bind.annotation.XmlSchema
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
        ... 5 more
```

I tried various fixes found online but none of them seems to work.

**Comment by karianna** [comment, marked as answer]
@mattiascibien Does this project work on Java 8 but not Java 11?  Are you able to also try with a build from AdoptOpenJDK.net?

We ask because this looks like the typical challenge when upgrading from Java 8 to 11 (where the JAXB implementation was removed from the JDK and you need to reference it via a 3rd party library instead.  See https://docs.microsoft.com/en-us/azure/developer/java/fundamentals/transition-from-java-8-to-java-11.

**Reply by mattiascibien** [reply]
I used maui-check to install all dependency and it downloads microsoft JDK for it. I do not think android SDK is compatible with that.

**Reply by mattiascibien** [reply]
The strange thing is that MAUI-check stil wants to use Mcirosft OpenJDK 11

**Reply by brunoborges** [reply]
Could you please provide us with the steps to reproduce the issue?

What kind of application are you building? Which IDE are you using? What were you using before the MS Build of OpenJDK? Which JDK were you using until March?

**Reply by mattiascibien** [reply]
I am on Windows 10. No JDK before.

I just run the maui-check program to install the required dependencies for using MAUI (after installing openjdk manually). During the creation of the Android Emulator it crashes with that exception (the android emulator creation process, SDK installing and license accepting, strangely, are working fine)

Here the link of the program: https://github.com/Redth/dotnet-maui-check.

I have found a guide on medium on how to add some external libs in order to fill the missing one on JDK 11 but I cannot find it ATM as I am on my work PC. A great idea could be to include them already in the OpenJDK 11 as they seem to be required to use them for the Android SDK.

**Reply by brunoborges** [reply]
Okz seems like you are developing a Xamarin/MAUI application with Visual Studio, and targeting Android.

For this scenario, we suggest you to install the JDK recommended by the Xamarin/MAUI documentation.

As far as I know, Android does not support Java 11 - which is the version of the MS Build of OpenJDK.

**Reply by mattiascibien** [reply]
Thanks for your support. I am going to open an issue to the developer of maui-check

**Reply by mattiascibien** [reply]
@brunoborges according to https://github.com/dotnet/net6-mobile-samples/blob/17b732a672926fb22f06739f0679a9b68d170810/README.md#android Microsoft OpenJDK 11 is the recommended version for using MAUI. Am I missing something?

**Reply by brunoborges** [reply]
This is an issue with how MAUI sets the JDK and the required libraries in its classpath. OpenJDK 11 (all vendors, not only Microsoft's) no longer ships an XML parser.

Please do create an issue with the net6-mobile project.

**Reply by brunoborges** [reply]
@mattiascibien let us know once you submit the issue to the net6-mobile project. A link to the issue will be appreciated!
